src/Components/ImagesTable/ImageLink.js

```diff
@@ -1,126 +1,49 @@
-import React from 'react';
+import React, { Suspense } from 'react';
 import PropTypes from 'prop-types';
-
-import {
-  Button,
-  Popover,
-  Text,
-  TextContent,
-  TextVariants,
-} from '@patternfly/react-core';
-import { DownloadIcon, ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { Button } from '@patternfly/react-core';
+import { useLoadModule, useScalprum } from '@scalprum/react-core';
+import ImageLinkDirect from './ImageLinkDirect';
 
 const ImageLink = (props) => {
-  const fileExtensions = {
-    vsphere: '.vmdk',
-    'guest-image': '.qcow2',
-    'image-installer': '.iso',
-  };
+  const scalprum = useScalprum();
+  const hasProvisionig = scalprum.initialized && scalprum.config?.provisioning;
+  if (hasProvisionig && props.imageType === 'ami') {
+    const [wizardOpen, openWizard] = React.useState(false);
+    const [{ default: ProvisioningWizard }, error] = useLoadModule(
+      {
+        appName: 'provisioning', // optional
+        scope: 'provisioning',
+        module: './ProvisioningWizard',
+        // processor: (val) => val, // optional
+      },
+      {},
+      {}
+    );
 
-  const uploadStatus = props.imageStatus
-    ? props.imageStatus.upload_status
-    : undefined;
-  if (uploadStatus) {
-    if (uploadStatus.type === 'aws') {
-      const url =
-        'https://console.aws.amazon.com/ec2/v2/home?region=' +
-        uploadStatus.options.region +
-        '#LaunchInstanceWizard:ami=' +
-        uploadStatus.options.ami;
-      return (
-        <Button
-          component="a"
-          target="_blank"
-          variant="link"
-          icon={<ExternalLinkAltIcon />}
-          iconPosition="right"
-          isInline
-          href={url}
-        >
-          Launch instance
-        </Button>
-      );
-    } else if (uploadStatus.type === 'azure') {
-      const url =
-        'https://portal.azure.com/#@' +
-        props.uploadOptions.tenant_id +
-        '/resource/subscriptions/' +
-        props.uploadOptions.subscription_id +
-        '/resourceGroups/' +
-        props.uploadOptions.resource_group +
-        '/providers/Microsoft.Compute/images/' +
-        uploadStatus.options.image_name;
+    if (!error) {
```

Comment:
Wait now all the conditions are gone? Will we get an error if it's the wrong image type or stage-stable?

Reply:
it's double `if` the upper condition gates by existence of provisioning app in the environment, if the environment lacks provisioning, we'd error out on `useLoadModule`, so that needs to be gated by the provisioning existence.

Reply:
oh that's scalprum? (unfamiliar with it).